management/src/main/java/org/ehcache/management/registry/DefaultCollectorService.java

```diff
@@ -0,0 +1,215 @@
+            new ContextualNotification(
+                configuration.getContext(),
+                EhcacheNotification.CACHE_MANAGER_CLOSED.name())));
+
+        // deregister me
+        cacheManagerProviderService.getCacheManager().deregisterListener(this);
+        break;
+    }
+  }
+
+  @Override
+  public void startStatisticCollector() {
+    if (collectingStats.compareAndSet(false, true)) {
+      StatisticsProviderConfiguration providerConfiguration = configuration.getConfigurationFor(EhcacheStatisticsProvider.class);
+
+      long timeToDisableMs = TimeUnit.MILLISECONDS.convert(providerConfiguration.timeToDisable(), providerConfiguration.timeToDisableUnit());
+      long pollingIntervalMs = Math.round(timeToDisableMs * 0.75); // we poll at 75% of the time to disable (before the time to disable happens)
+      final AtomicLong lastPoll = new AtomicLong(System.currentTimeMillis());
+
+      task = scheduledExecutorService.scheduleWithFixedDelay(new Runnable() {
+        @Override
+        public void run() {
+          try {
+            CacheManager cacheManager = cacheManagerProviderService.getCacheManager();
+
+            // always check if the cache manager is still available
+            if (cacheManager.getStatus() == Status.AVAILABLE) {
+
+              // create the full context list from current caches
+              Collection<Context> cacheContexts = new ArrayList<Context>();
+              for (ContextContainer cacheContext : managementRegistry.getContextContainer().getSubContexts()) {
+                cacheContexts.add(configuration.getContext().with(cacheContext.getName(), cacheContext.getValue()));
+              }
+
+              long now = timeSource.getTimeSource().getTimeMillis();
+              Collection<ContextualStatistics> statistics = new ArrayList<ContextualStatistics>();
+
+              // for each capability, call the management registry
+              for (Map.Entry<String, StatisticQuery.Builder> entry : selectedStatsPerCapability.entrySet()) {
+                for (ContextualStatistics contextualStatistics : entry.getValue().since(lastPoll.get()).on(cacheContexts).build().execute()) {
+                  statistics.add(contextualStatistics);
+                }
+              }
+
+              // next time, only poll history from this time
+              lastPoll.set(System.currentTimeMillis());
+
+              if (!statistics.isEmpty()) {
+                messageConsumer.accept(new DefaultMessage(now, statistics.toArray(new ContextualStatistics[statistics.size()])));
+              }
+            }
+          } catch (RuntimeException e) {
+            LOGGER.error(e.getMessage(), e);
+          }
+        }
+      }, pollingIntervalMs, pollingIntervalMs, TimeUnit.MILLISECONDS);
+    }
+  }
+
+  @Override
+  public void stopStatisticCollector() {
+    if (collectingStats.compareAndSet(true, false)) {
+      task.cancel(false);
+      task = null;
```

Comment:
Thanks! Will do!

Note: would synchronized work also ? because I am wondering if I also need to make sure that start / stop / stopCollector and startCollector are exclusive ?